Bug: https://github.com/openzfs/zfs/pull/15651
Forwarded: yes
Applied-Upstream: master
Last-Update: 2024-01-09
Author: Mauricio Faria de Oliveira <mfo@canonical.com>
Description: [PATCH] zed: fix typo in variable
 ZED_POWER_OFF_ENCLO*US*RE_SLOT_ON_FAULT
 .
 Replace ENCLO_US_RE with ENCLO_SU_RE in the name of the variable.
 .
 Note this changes the user-visible string in zed.rc, thus might
 break current users with the wrong string, but it's ~2 months
 since zfs-2.2.0 tag is out, thus should not be widespread yet.

Index: zfs-linux/cmd/zed/zed.d/statechange-slot_off.sh
=======================================================
--- zfs-linux.orig/cmd/zed/zed.d/statechange-slot_off.sh
+++ zfs-linux/cmd/zed/zed.d/statechange-slot_off.sh
@@ -5,7 +5,7 @@
 #
 # Bad SCSI disks can often "disappear and reappear" causing all sorts of chaos
 # as they flip between FAULTED and ONLINE.  If
-# ZED_POWER_OFF_ENCLOUSRE_SLOT_ON_FAULT is set in zed.rc, and the disk gets
+# ZED_POWER_OFF_ENCLOSURE_SLOT_ON_FAULT is set in zed.rc, and the disk gets
 # FAULTED, then power down the slot via sysfs:
 #
 # /sys/class/enclosure/<enclosure>/<slot>/power_status
@@ -19,7 +19,7 @@
 # Exit codes:
 #   0: slot successfully powered off
 #   1: enclosure not available
-#   2: ZED_POWER_OFF_ENCLOUSRE_SLOT_ON_FAULT disabled
+#   2: ZED_POWER_OFF_ENCLOSURE_SLOT_ON_FAULT disabled
 #   3: vdev was not FAULTED
 #   4: The enclosure sysfs path passed from ZFS does not exist
 #   5: Enclosure slot didn't actually turn off after we told it to
@@ -32,7 +32,7 @@ if [ ! -d /sys/class/enclosure ] ; then
 	exit 1
 fi
 
-if [ "${ZED_POWER_OFF_ENCLOUSRE_SLOT_ON_FAULT}" != "1" ] ; then
+if [ "${ZED_POWER_OFF_ENCLOSURE_SLOT_ON_FAULT}" != "1" ] ; then
 	exit 2
 fi
 
Index: zfs-linux/cmd/zed/zed.d/zed.rc
=======================================================
--- zfs-linux.orig/cmd/zed/zed.d/zed.rc
+++ zfs-linux/cmd/zed/zed.d/zed.rc
@@ -146,4 +146,4 @@ ZED_SYSLOG_SUBCLASS_EXCLUDE="history_event"
 # Power off the drive's slot in the enclosure if it becomes FAULTED.  This can
 # help silence misbehaving drives.  This assumes your drive enclosure fully
 # supports slot power control via sysfs.
-#ZED_POWER_OFF_ENCLOUSRE_SLOT_ON_FAULT=1
+#ZED_POWER_OFF_ENCLOSURE_SLOT_ON_FAULT=1
